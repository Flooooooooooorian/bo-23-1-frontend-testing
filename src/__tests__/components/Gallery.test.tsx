import {render} from "@testing-library/react";
import Gallery from "../../components/Gallery";
import {Item} from "../../models";

jest.mock("../../components/ItemCard", () => (props: any) => <div>{props.item.name + props.item.id}</div>)

describe("Gallery Tests", () => {
    test("Snapshot", () => {

        const testItems: Item[] = [{id: "1", name: "item-1"}, {id: "2", name: "item-2"}]

        const view = render(<Gallery items={testItems}/>)

        expect(view.asFragment()).toMatchSnapshot()
    })
})
