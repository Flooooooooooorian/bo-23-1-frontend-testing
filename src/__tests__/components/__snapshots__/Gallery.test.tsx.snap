// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Tests Snapshot 1`] = `
<DocumentFragment>
  <h1>
    Item-Gallery
  </h1>
  <div>
    item-11
  </div>
  <div>
    item-22
  </div>
</DocumentFragment>
`;
